xwiki-platform-core/xwiki-platform-csrf/src/main/java/org/xwiki/csrf/internal/DefaultCSRFToken.java

```diff
@@ -193,9 +197,14 @@ public String getResubmissionURL()
 
         // back URL is the URL of the document that was about to be modified, so in most
         // cases we can redirect back to the correct document (if the user clicks "no")
-        String backUrl = getDocumentURL(this.docBridge.getCurrentDocumentReference(), null);
+        DocumentReference currentDocRef = this.docBridge.getCurrentDocumentReference();
+        // If the current document is null (on the login page for example), let's redirect to the default page
+        if (currentDocRef == null) {
+            currentDocRef = resolver.resolve("");
+        }
+        String backUrl = getDocumentURL(currentDocRef, null);
         query += "&xback=" + urlEncode(backUrl);
-
+        
```

Comment:
I know this kind of changes are easy to introduce by accident, but the PRs and commit diffs are easier to read when there is no other changes than the one strictly related to the PR :)